sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -219,39 +264,45 @@ export interface SendableMessageInfo {
     messageId?: string | number | Buffer;
     partitionKey?: string;
+    properties?: {
+        [key: string]: any;
+    };
     replyTo?: string;
     replyToSessionId?: string;
     scheduledEnqueueTimeUtc?: Date;
     sessionId?: string;
     timeToLive?: number;
     to?: string;
-    userProperties?: {
-        [key: string]: any;
-    };
     viaPartitionKey?: string;
 }
 
 // @public
 export class Sender {
-    cancelScheduledMessage(sequenceNumber: Long): Promise<void>;
-    cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void>;
+    cancelScheduledMessage(sequenceNumber: Long, options?: OperationOptions): Promise<void>;
+    cancelScheduledMessages(sequenceNumbers: Long[], options?: OperationOptions): Promise<void>;
     close(): Promise<void>;
     get isClosed(): boolean;
-    scheduleMessage(scheduledEnqueueTimeUtc: Date, message: SendableMessageInfo): Promise<Long>;
-    scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: SendableMessageInfo[]): Promise<Long[]>;
-    send(message: SendableMessageInfo): Promise<void>;
-    sendBatch(messages: SendableMessageInfo[]): Promise<void>;
+    scheduleMessage(scheduledEnqueueTimeUtc: Date, message: SendableMessageInfo, options?: OperationOptions): Promise<Long>;
+    scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: SendableMessageInfo[], options?: OperationOptions): Promise<Long[]>;
+    send(message: SendableMessageInfo, options?: OperationOptions): Promise<void>;
+    sendBatch(messages: SendableMessageInfo[], options?: OperationOptions): Promise<void>;
     }
 
-// @public
+// @public (undocumented)
 export class ServiceBusClient {
     constructor(connectionString: string, options?: ServiceBusClientOptions);
-    constructor(host: string, credential: TokenCredential, options?: ServiceBusClientOptions);
-    close(): Promise<any>;
-    createQueueClient(queueName: string): QueueClient;
-    createSubscriptionClient(topicName: string, subscriptionName: string): SubscriptionClient;
-    createTopicClient(topicName: string): TopicClient;
-    readonly name: string;
+    constructor(hostName: string, tokenCredential: TokenCredential_2, options?: ServiceBusClientOptions);
+    // (undocumented)
+    close(): Promise<void>;
+    createReceiver(queueName: string, receiveMode: "peekLock"): NonSessionReceiver<"peekLock">;
+    createReceiver(queueName: string, receiveMode: "receiveAndDelete"): NonSessionReceiver<"receiveAndDelete">;
+    createReceiver(topicName: string, subscriptionName: string, receiveMode: "peekLock"): NonSessionReceiver<"peekLock"> & SubscriptionRuleManagement;
+    createReceiver(topicName: string, subscriptionName: string, receiveMode: "receiveAndDelete"): NonSessionReceiver<"receiveAndDelete"> & SubscriptionRuleManagement;
+    createSender(queueOrTopicName: string): Sender;
+    createSessionReceiver(queueName: string, receiveMode: "peekLock", sessionId: string | "", options?: CreateSessionReceiverOptions): SessionReceiver<"peekLock">;
```

Comment:
In .Net, the creation of the session receiver is an async operation as it covers the creation of the link as well. 
cc @JoshLove-msft for details, but we should sync on this

Reply:
Agreed and we should.

I believe that they were specifically trying to cover the "server gives me next unlocked session" since this isn't an issue if you actually name one. 

We might actually be able to do type stuff to make that more apparent.